server/gitrest/src/routes/repository/commits.ts

```diff
@@ -7,48 +7,73 @@ import * as resources from "@fluidframework/gitresources";
+import * as winston from "winston";
+import { ExternalStorageManager } from "../../ExternalStorageManager";
 import * as utils from "../../utils";
 
 export async function getCommits(
     repoManager: utils.RepositoryManager,
     owner: string,
     repo: string,
-    ref: string,
-    count: number): Promise<resources.ICommitDetails[]> {
+    count: number,
+    externalStorageManager: ExternalStorageManager): Promise<resources.ICommitDetails[]> {
     const repository = await repoManager.open(owner, repo);
-    const walker = git.Revwalk.create(repository);
+    try {
+        const walker = git.Revwalk.create(repository);
 
-    // eslint-disable-next-line no-bitwise
-    walker.sorting(git.Revwalk.SORT.TOPOLOGICAL | git.Revwalk.SORT.TIME);
+        // eslint-disable-next-line no-bitwise
+        walker.sorting(git.Revwalk.SORT.TOPOLOGICAL | git.Revwalk.SORT.TIME);
 
-    // Lookup the commits specified from the given revision
-    const revObj = await git.Revparse.single(repository, ref);
-    walker.push(revObj.id());
-    const commits = await walker.getCommits(count);
+        // Lookup the commits specified from the given revision
+        const revObj = await git.Revparse.single(repository, ref);
+        walker.push(revObj.id());
+        const commits = await walker.getCommits(count);
 
-    const detailedCommits = commits.map(async (rawCommit) => {
-        const gitCommit = await utils.commitToICommit(rawCommit);
-        const result: resources.ICommitDetails =
-        {
-            commit: {
-                author: gitCommit.author,
-                committer: gitCommit.committer,
-                message: gitCommit.message,
-                tree: gitCommit.tree,
-                url: gitCommit.url,
-            },
-            parents: gitCommit.parents,
-            sha: gitCommit.sha,
-            url: "",
+        const detailedCommits = commits.map(async (rawCommit) => {
+            const gitCommit = await utils.commitToICommit(rawCommit);
+            const result: resources.ICommitDetails =
+            {
+                commit: {
+                    author: gitCommit.author,
+                    committer: gitCommit.committer,
+                    message: gitCommit.message,
+                    tree: gitCommit.tree,
+                    url: gitCommit.url,
+                },
+                parents: gitCommit.parents,
+                sha: gitCommit.sha,
+                url: "",
 
-        };
-        return result;
-    });
+            };
+            return result;
+        });
 
-    return Promise.all(detailedCommits);
+        return Promise.all(detailedCommits);
+    } catch (err) {
+        if (process.env.EXTERNAL_STORAGE_ENABLED != "true") {
+            winston.info(`External storage is not enabled`);
+            return;
+        } else {
+            if (config.get("externalStorage:endpoint") != null)
+            {
+                winston.error("getCommits error: " + err);
```

Comment:
Should we be logging this as an error closer to where we're actually throwing it as an error (i.e. lines 67-68)?

Reply:
I see your point, actually I think this log should be there as info regardless of whether external storage is enabled. I will change this to info and move it to when we catch the error. Also will add more detail to the other logs.